Public/Src/Utilities/KeyValueStore/IKeyValueStore.cs

```diff
@@ -82,5 +82,21 @@ public interface IKeyValueStore<in TKey, TValue>
         /// otherwise, false.
         /// </returns>
         bool Contains(TKey key, string columnFamilyName = null);
+
+        /// <summary>
+        /// Applies a batch Put and Delete operations. Useful only in the case when a large number of operations have 
+        /// to be performed on the store. May or may be atomic.
+        /// </summary>
+        /// <param name="keys">
+        /// The key of each entry.
+        /// </param>
+        /// <param name="values">
+        /// The value of each entry. Must be the same size as the keys. If a value is null, then a Delete will be 
+        /// performed; otherwise, a Put.
+        /// </param>
+        /// <param name="columnFamilyName">
+        /// The column family to use.
+        /// </param>
+        void ApplyBatch(IEnumerable<TKey> keys, IEnumerable<TValue> values, string columnFamilyName = null);

```

Comment:
>IEnumerable<TKey> keys, IEnumerable<TValue> values [](start = 24, length = 50)

Just use IEnumerable<KeyValuePair<TKey, TValue>>?

Reply:
Tried this at the beginning; but its not possible: the type is contravariant on TKey

---
In reply to: [292134200](https://github.com/microsoft/BuildXL/pull/403#discussion_r292134200) [](ancestors = 292134200)

Reply:
Does a tuple work?

---
In reply to: [292149253](https://github.com/microsoft/BuildXL/pull/403#discussion_r292149253) [](ancestors = 292149253,292134200)

Reply:
No, it has the same issue